server/models/Transaction.js

```diff
@@ -439,51 +440,82 @@ export default (Sequelize, DataTypes) => {
-        transaction.CreatedByUserId = CreatedByUserId;
-        transaction.FromCollectiveId = FromCollectiveId;
-        transaction.CollectiveId = CollectiveId;
-        transaction.PaymentMethodId = transaction.PaymentMethodId || PaymentMethodId;
-        transaction.type = transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT;
-        transaction.platformFeeInHostCurrency = toNegative(transaction.platformFeeInHostCurrency);
-        transaction.hostFeeInHostCurrency = toNegative(transaction.hostFeeInHostCurrency);
-        transaction.taxAmount = toNegative(transaction.taxAmount);
-        transaction.paymentProcessorFeeInHostCurrency = toNegative(transaction.paymentProcessorFeeInHostCurrency);
-
-        if (transaction.amount > 0) {
-          // populate netAmountInCollectiveCurrency for donations
-          const fees =
-            (transaction.taxAmount || 0) +
-            transaction.platformFeeInHostCurrency +
-            transaction.hostFeeInHostCurrency +
-            transaction.paymentProcessorFeeInHostCurrency;
-          transaction.netAmountInCollectiveCurrency = transaction.amountInHostCurrency + fees; // `fees` is a negative number
-          if (transaction.hostCurrencyFxRate) {
-            transaction.netAmountInCollectiveCurrency = Math.round(
-              transaction.netAmountInCollectiveCurrency / transaction.hostCurrencyFxRate,
-            );
-          }
-        }
-        return Transaction.createDoubleEntry(transaction);
-      });
+    const collective = await models.Collective.findByPk(CollectiveId);
+    const HostCollectiveId = await collective.getHostCollectiveId();
+    if (!HostCollectiveId && !transaction.HostCollectiveId) {
+      throw new Error(`Cannot create a transaction: collective id ${CollectiveId} doesn't have a host`);
+    }
+    transaction.HostCollectiveId = HostCollectiveId || transaction.HostCollectiveId;
+    // attach other objects manually. Needed for afterCreate hook to work properly
+    transaction.CreatedByUserId = CreatedByUserId;
+    transaction.FromCollectiveId = FromCollectiveId;
+    transaction.CollectiveId = CollectiveId;
+    transaction.PaymentMethodId = transaction.PaymentMethodId || PaymentMethodId;
+    transaction.type = transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT;
+    transaction.hostFeeInHostCurrency = toNegative(transaction.hostFeeInHostCurrency);
+    transaction.platformFeeInHostCurrency = toNegative(transaction.platformFeeInHostCurrency);
+    transaction.taxAmount = toNegative(transaction.taxAmount);
+    transaction.paymentProcessorFeeInHostCurrency = toNegative(transaction.paymentProcessorFeeInHostCurrency);
+    // Separate donation transaction and remove platformFee from main transaction
+    if (transaction.data?.isFeesOnTop) {
+      const platformCurrencyFxRate = await getFxRate(transaction.currency, FEES_ON_TOP_TRANSACTION_PROPERTIES.currency);
```

Comment:
Can you extract that block out in another model (what is specific to isFeesOnTop) to make it clearer?